### ISSUE TITLE
New package for inclusion - alanning:roles

### ISSUE THREAD

**Issue by StorytellerCZ** [Meteor-Community-Packages/organization#25]
## Package/project name & description
`alanning:roles` is the recommended package when it comes to roles.

## Links
* [Atmosphere](https://atmospherejs.com/alanning/roles)
* [Github repository](https://github.com/alanning/meteor-roles)

## Current status of the project
 - [x] Abandoned

 ## Reasoning
 
Alanning's roles is the recommended package for managing roles in Meteor. Even though major work on version 2 has been done, the package has been abandoned even though there are PRs.

@mitar Any news about this package and what can be done to get v2?

**Comment by mitar** [comment]
Background: see discussion here: https://github.com/alanning/meteor-roles/issues/270#issuecomment-533396511

In fact there is more development here. I think we should (if @alanning agrees):
* Move roles to this repository.
* Release v2 with my changes.
* Make v3 with what @SimonSimCity made.

**Comment by StorytellerCZ** [comment]
@mitar All for this plan.

**Comment by alanning** [comment]
Sounds good to me.  I'll update the readme on the alanning/meteor-roles repo to link here once this is ready.

(Although I personally don't consider roles v1 abandoned I haven't done a good job of communicating current status.  What's the proper term for security fixes only?)

**Comment by coagmano** [comment]
>  What's the proper term for security fixes only?
Long Term Support / LTS?

**Comment by StorytellerCZ** [comment]
[Repo status](https://www.repostatus.org/) seems pretty good, but in this case it is missing the LTS tag.

**Comment by mitar** [comment]
@alanning Just to be clear, I am proposing that:
* We move the whole repository under this organization.
* We can keep both v1, v2, and v3 branches maintained (probably v3 would be the only one getting feature updates, but we could keep v1 and v2 with security/compatibility updates).
* We would be publishing them under the same Meteor package name.

If you agree with that, could you: add me as an owner to the repository, and add `communitypackages` as maintainer of Meteor package:

```
meteor admin maintainers alanning:roles --add communitypackages`
```

**Comment by alanning** [comment]
@mitar OK, sounds good.  

**Add maintainer**

```
$ meteor admin maintainers alanning:roles --add communitypackages
Adding a maintainer to alanning:roles...

The maintainers for alanning:roles are:
alanning
mitar
communitypackages
```

**Repo ownership**
I think github repos can only have one "owner".  You are already a "Collaborator" in github for alanning/meteor-roles; is that sufficient?  Perhaps I should use the "Transfer ownership" option to move the repo to "Meteor-Community-Packages"?

**Comment by mitar** [comment]
I think you can make me an `Admin` which would then allow me to transfer ownership. I think currently I have only `Write` permissions there. I do not think you can transfer ownership because without having permissions on this organization? But you can try.

**Comment by alanning** [comment]
Yep, you're correct.  Transfer attempt fails with, "You don’t have the permission to create public repositories on Meteor-Community-Packages".

Personal github repos are a [bit different](https://help.github.com/en/articles/access-permissions-on-github#personal-user-accounts) than organization ones so either I'll need to be given "create repo" permissions on the org or I'll need to transfer ownership of the repo to you first and then you can transfer to the org.

In order for me to transfer the repo over to you, you will first need to delete your existing mitar/meteor-roles repo.

**Comment by mitar** [comment]
> In order for me to transfer the repo over to you, you will first need to delete your existing mitar/meteor-roles repo.

Done.

**Comment by copleykj** [comment]
@alanning If you wanted, we'd be more than happy to have you as a member of this org as well

**Comment by alanning** [comment]
@copleykj Thank you for the invite.  Yes, that sounds good.  Will get ownership transferred over and read through the org docs.

**Comment by StorytellerCZ** [comment]
Created a group for the project.

**Comment by alanning** [comment]
Repo transferred!

**Comment by mitar** [comment]
Awesome. I suggest that we make the current `master` branch into `v1.0` branch. And then then @SimonSimCity pushes his stuff into `master` branch as 3.0.

**Comment by mitar** [comment]
I added @SimonSimCity to the team of maintainers for the repo.

**Comment by mitar** [comment]
TODO: We have also to create a Meteor org of maintainers of the package.

**Comment by SimonSimCity** [comment]
@mitar I'd propose that we also tag a `2.0` version on master where the current `v2.0` branch is just merged in. I'll also take a look at the issues and see how we should proceed with them. Most of them will just be asked to move up to a newer version. Should we also leave some "open space" for bugfix versions or will everybody simply have to update to `3.0`?

**Comment by mitar** [comment]
> I'd propose that we also merge the current v2.0 branch into master and tag it as 2.0.

So then we first merge v2.0 in to master, and then merge v3.0 into master? I could be OK with that, if you built v3.0 out of v2.0?

I would not use tags in this particular case though, because it seems @alanning would prefer if all versions are maintained.

> Should we also leave some "open space" for bugfix versions or will everybody simply have to update to 3.0?

I think if they are just bugfixes (even security ones) we should do them. If they are feature requests, we should suggest to move to 3.0.

**Comment by SimonSimCity** [comment]
My version (v3) is a modification on top of v2, yes.

Ok, so how to we realize this when all are merged into master? I originally thought about an approach like gitflow, but I have no idea how we could then realize the work of maintaining all old versions.

Another approach of mine is that we merge everything into master and if v1 needs a bugfix, we create a separate v1 branch which will never be merged back but will flow "on it's own" and some of the commits there might then get a tag if they're released as version 1.1 or so. Is this an approach everyone could live with? All this requires that Atmosphere allows us to release a v1.1 after v2 has been uploaded - I've never tried it.

**Comment by SimonSimCity** [comment]
Another approach would be the one I've seen in TYPO3. They had the normal development done in master and everytime they were ready to release, they created a branch from which the release was following off from. Every change done for a release was then merged back into `master` where also the development was.

How does that sound? It's a completely different model but allows for easily maintaining multiple versions.

**Comment by sebakerckhof** [comment]
> All this requires that Atmosphere allows us to release a v1.1 after v2 has been uploaded - I've never tried it.

Yep this works.

**Comment by StorytellerCZ** [comment]
Since the repo has been transferred and everything set, I'm closing this issue. The discussion I think should continue in the repo with actual PRs.

**Comment by mitar** [comment]
Started https://github.com/Meteor-Community-Packages/meteor-roles/issues/283.
